content/news/XXXX-XX-XX-dvs-internals-pt3.md

```diff
@@ -0,0 +1,219 @@
+
+The carrier wave operates at **2500 Hz**, a significant increase from Serato’s
+**1000 Hz**.
+
+> **Advantage:** The higher carrier frequency allows for 2.5× greater resolution.
+
+![Offset-modulated Signal]({static}/images/news/traktor-mk2-signal.svg)
+
+Upon inspection, this waveform clearly doesn’t use amplitude modulation - the
+amplitude remains constant. Instead, it appears to be **offset-modulated**,
+where the signal is shifted vertically from the x-axis. This is a non-standard
+technique not commonly used in typical modulation schemes.
+
+On the original vinyl version (not shown here due to copyright), the offset
+can be so large that the signal floats entirely above the x-axis for multiple
+cycles - making zero-crossing detection impossible.
+
+Even when that doesn’t happen, the offset causes the time interval $\Delta t$
+between zero-crossings to become irregular, introducing audible pitch flutter.
+
+To decode the signal, we must solve:
+
+1. How to filter the signal to enable pitch detection
+2. How to demodulate this non-standard modulation
+3. How to decipher the code that is modulated onto the carrier
+
+---
+
+## Pitch Detection
+
+> **Note:** If you're unfamiliar with pitch detection in DVS systems, revisit
+> [DVS Internals Pt. 1]({filename}/news/2021-11-21-dvs-internals-pt1.md).
+
+Our goal is to produce a signal that oscillates evenly around the x-axis. This
+filtered waveform can then be processed by the existing pitch detection algorithm.
+
+A simple discrete derivative operation achieves this:
+
+$$
+y[n] = x[n] - x[n-1] \tag{1}
+$$
+
+$\text{where:}$<br>
+$\text{- x[n]: Input sample}$<br>
+$\text{- x[n-1]: Delayed input sample}$<br>
+$\text{- y[n]: Difference of both values}$<br>
+<br>
+
+When applied to the offset-modulated signal, we get:
+
+![Traktor MK2 signal with derivative]({static}/images/news/traktor-mk2-signal-with-derivative.svg)
+
+The resulting waveform oscillates cleanly around zero, which is ideal for
+analysis. It also makes it easier to pinpoint the half-cycle peaks needed for
+bit detection.
+
+---
+
+## Demodulation Techniques
+
+To extract bits from the signal, we detect the zero-crossings and sample the
+amplitude of the sinusoid at those moments.
+
+![Traktor MK2 signal with
+zero-crossings]({static}/images/news/traktor-mk2-signal-with-zero-crossings.svg)
+
+You may notice that the derivative’s zero-crossings don’t align perfectly with
+the original peaks. That’s due to a delay introduced by the filter. Smoothing
+the signal first, then compensating for the delay (e.g., by selecting
+$x[n-3]$), yields better results.
+
+For greater accuracy, one could analyze the phase response $\phi(\omega)$,
+which shows how filter delay varies with input frequency - but for this use case,
+a fixed delay works well enough.
+
+The filtered signal can cross the x-axis in two directions-positive to
+negative or vice versa. Based on the direction, we determine which half-cycle
+contains the encoded bit.
```

Comment:
I assume the amplitude is only sampled when the filtered signal crosses from negative-to-positive? At least thats my conclusion from the graph, it would nice if this was also clarified in text. Is the info from the other changes/directions just ignored? Couldn't it be used as redundant info useful for error correction?